### PR TITLE
🔨 refactor `!find` command

### PR DESCRIPTION
**Pull request by idinium96**
### Old:
-    Problem:
    - only show 100 entries if too many.
    - show in `/code` and entry form will cause the chat window to lag
-    Screenshot:
image

### New:
-    Only show `sku - name`
-    Can list all and no lagging issue.
image
.
.
.
image

### CODE REVIEW

File: src/classes/Commands/sub-classes/PricelistManager.ts [hunk 1]
```diff
@@ -1288,50 +1288,27 @@ export default class PricelistManagerCommands {
...
-                    `/code ${this.generateOutput(filter, true, 80, filterCount > 100 ? 100 : filterCount)}`
-                );
+
+            const listCount = list.length;
+            if (listCount > 200) {
+                const loops = Math.ceil(listCount / 200);
+
+                for (let i = 0; i < loops; i++) {
+                    const last = loops - i === 1;
+                    await sleepasync().Promise.sleep(1 * 1000);
```
Comment (arik123) [inline src/classes/Commands/sub-classes/PricelistManager.ts:1303]: put this line after first iteration

File: src/classes/Commands/sub-classes/PricelistManager.ts [hunk 2]
```diff
@@ -1288,50 +1288,27 @@ export default class PricelistManagerCommands {
...
-                await sleepasync().Promise.sleep(1 * 1000);
-                this.bot.sendMessage(steamID, `/code ${this.generateOutput(filter, true, 60, 80)}`);
-                await sleepasync().Promise.sleep(1 * 1000);
-                this.bot.sendMessage(
-                    steamID,
-                    `/code ${this.generateOutput(filter, true, 80, filterCount > 100 ? 100 : filterCount)}`
-                );
+
+            const listCount = list.length;
+            if (listCount > 200) {
```
Comment (arik123) [inline src/classes/Commands/sub-classes/PricelistManager.ts:1298]: this path should be fine for any listing count

**Review by arik123 — CHANGES_REQUESTED** [review]
There is no need for 2 different codepaths when sending messages,
put sleep after first message - faster response
I think that some kind of limit would be usefull, maybe add option `limit=<value>` and if it is set to -1 then it would list all. Also set some good default limit 50-200.

File: src/classes/Commands/sub-classes/PricelistManager.ts (hunk 2 above)
Comment (idinium96) [reply src/classes/Commands/sub-classes/PricelistManager.ts:1298]: I see. So no need to do that if statement. got it. thanks

**Comment by idinium96** [comment]
Lasted commit:
- Default limit = 200
- if result < 200:
image

- if result > 200 and limit not defined:
image

- if result > 200 and limit is defined:
image
image
image

**Review by arik123 — APPROVED** [review]

**Pull request merged**